content/en/tracing/guide/tutorial-enable-python-container-agent-host.md

```diff
@@ -0,0 +1,422 @@
+{{< /code-block >}}
+
+   The application is ready to use when you see the following output in the terminal: 
+
+   ```
+   notes          |  * Debug mode: on
+   notes          | INFO:werkzeug:WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
+   notes          |  * Running on all addresses (0.0.0.0)
+   notes          |  * Running on http://127.0.0.1:8080
+   notes          |  * Running on http://192.168.32.3:8080
+   notes          | INFO:werkzeug:Press CTRL+C to quit
+   notes          | INFO:werkzeug: * Restarting with stat
+   notes          | WARNING:werkzeug: * Debugger is active!
+   notes          | INFO:werkzeug: * Debugger PIN: 143-375-699
+   ```
+
+   You can also verify that it's running by viewing the running containers with the `docker ps` command.
+
+3. Open up another terminal and send API requests to exercise the app. The notes application is a REST API that stores data in a Postgres database running in another container. Send it a few commands:
+
+`curl -X GET 'localhost:8080/notes'`
+: `{}`
+
+`curl -X POST 'localhost:8080/notes?desc=hello'`
+: `(1, hello)`
+
+`curl -X GET 'localhost:8080/notes?id=1'`
+: `(1, hello)`
+
+`curl -X GET 'localhost:8080/notes'`
+: `{”1”, "hello"}`
+
+`curl -X PUT 'localhost:8080/notes?id=1&desc=UpdatedNote'`
+: `(1, UpdatedNote)`
+
+`curl -X DELETE 'localhost:8080/notes?id=1'`
+: `Deleted`
+
+### Stop the application
+
+After you've seen the application running, stop it so that you can enable tracing on it.
+
+1. Stop the containers:
+   {{< code-block lang="sh" >}}
+docker-compose -f docker/host-and-containers/exercise/docker-compose.yaml down
+{{< /code-block >}}
+
+2. Remove the containers:
+   {{< code-block lang="sh" >}}
+docker-compose -f docker/host-and-containers/exercise/docker-compose.yaml rm
+{{< /code-block >}}
+
+## Enable tracing
+
+Now that you have a working Python application, configure it to enable tracing.
+
+1. Add the Python tracing package to your project. Open the file `apm-tutorial-python/requirements.txt`, and add `ddtrace` to the list if it is not already there:
+
+   ```
+   flask==2.2.2
+   psycopg2-binary==2.9.3
+   requests==2.28.1
+   ddtrace
+   ```
+
+2. Within the notes application Dockerfile, `docker/host-and-containers/exercise/Dockerfile.notes`, change the CMD line that starts the application to use the `ddtrace` package:
+
+   ```
+   # Run the application with Datadog 
+   CMD ["ddtrace-run", "python", "-m", "notes_app.app"]
+   ```
+
+   This automatically instruments the application with Datadog services.
+
+3. Apply [Universal Service Tags][10], which identify traced services across different versions and deployment environments so that they can be correlated within Datadog, and you can use them to search and filter. The three environment variables used for Unified Service Tagging are `DD_SERVICE`, `DD_ENV`, and `DD_VERSION`. Add the following environment variables in the Dockerfile:
```

Comment:
```suggestion
3. Apply [Universal Service Tags][10], which identify traced services across different versions and deployment environments, so that they can be correlated within Datadog, and you can use them to search and filter. The three environment variables used for Unified Service Tagging are `DD_SERVICE`, `DD_ENV`, and `DD_VERSION`. Add the following environment variables in the Dockerfile:
```